components/html-block/html-block.js

```diff
@@ -275,6 +279,12 @@ class HtmlBlock extends LitElement {
 		await this._renderInline(e.target);
 	}
 
+	async _processEmbeds() {
+		const htmlFragment = document.createRange().createContextualFragment(this.html);
+		await renderEmbeds(htmlFragment);
+		return htmlFragment;
```

Comment:
Right now, `renderEmbeds` doesn't return anything; instead, it works on an existing DOM node. This makes it easy to adapt both to the full HTML document case and the html-block case, though that comes at the cost of relying on side effects in this case.